File: OOP C++/lab2/Emit.cpp
#include <stdexcept>
#include "FactoryInitializer.h"
#include "Emit.h"


void Emit::Execute(Operands& operands, Tokens& tokens, std::string& output, Reader& reader) {
    int operand1 = operands.GetAndPop();
    output +=  static_cast<char>(operand1);
    output += " ";
}

namespace {
    FactoryInitializer<Emit> Registration("emit");
}

File: Lab1/src/Commands/Emit.java
package Commands;

import Utility.Operands;

import java.util.StringTokenizer;

public class Emit implements Command{
    @Override
    public void Execute(StringTokenizer tokens, Operands operands) {
        int num1 = operands.GetAndPop();
        char ch = (char)num1;
        System.out.print(ch + " ");
    }

    @Override
    public void Check(StringTokenizer tokens, Operands operands) {

    }

}
